### ISSUE TITLE
Calling submit action does not trigger validation

### ISSUE THREAD

**Issue by LKay** [erikras/redux-form#3267]
### Are you submitting a **bug report** or a **feature request**?

This seems to be a bug.

### What is the current behavior?

When `submit` action is dispatched on form it just submits the data and totally ignores validation on either form or fields level.

### What is the expected behavior?

When `submit` action is dispatched then validaitons should be triggered.

### What's your environment?

`redux-form@7.0.3`

**Comment by erikras** [comment]
You might need to be using the [`<Form>`](http://redux-form.com/7.0.3/docs/api/Form.md/) component. Are you?

**Comment by LKay** [comment]
Yeah, I tried both regular `<form />` and `<Form />` from the `redux-form`. I have `redux-logger` attached and it seems than after dispatching `submit` action no validation is called (even though there are errors in the state) and imedietaly `FORM_SUBMIT_SUCCESS` action is being dispatched.

**Comment by erikras** [comment]
It seems to be working in the sandbox, right?

[![Edit Redux Form - Remote Submit Example](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/ElYvJR21K)

**Comment by LKay** [comment]
Yes, it seems to be working in sandbox. It seems to be also workin in my code but only if button is also inside a decorated component. If it is outside decorated context it just dispatches successful submit.

**Comment by danielrob** [comment]
@erikras adding a validation function to your sandbox: 

[![Edit Redux Form - Remote Submit Example](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/MQ2ZGNglO)

This does indeed appear to not run when submit action is dispatched.

**Comment by jarekb84** [comment]
Any resolution to this issue? We're looking at doing the same thing as @LKay where we have the submit buttons outside of the decorated form, and initiating a submit via dispatch of the redux-form submit action. However validation is not getting triggered in this manner. 

The [Form](https://redux-form.com/7.0.4/docs/api/form.md/) docs mention it should only be used if the submission is triggered from within the form component. So what is the correct approach to trigger validation on a remote submission?

Our current solution is to watch for changes in the form's invalid prop via componentWillReceiveProps and dispatch an action that disables the submit button in the parent component. It works, but does not feel like a clean way of handling this.

**Comment by dagda1** [comment]
Not being able to trigger validation leads to some really messy workarounds, especially with remote submits.  Would you be open to a PR that does trigger sync validation?  Is there a problem that I cannot see?

**Comment by theKashey** [comment]
There is no on-submit-validation, only on field change event.
But fact you can create an invalid form, and next submit it, before touching anything.
And it will be submitted.

**Comment by kennyx46** [comment]
Is this considered to be an issue? Is it going to be fixed?

**Comment by Osuriel** [comment]
yeah.l Im having the same issue. is there any updates?

**Comment by naniaryan** [comment]
is there any update on this issue ? or any workarounds ?

**Comment by caseybaggz** [comment]
Having this same issue with 2 out of 50 forms which is pretty odd.

**Comment by UlrikHelmers** [comment]
The issue seems to be present in earlier versions aswell. The problem is present in the actual example on 'Sync Validation' in version 6.8:
https://redux-form.com/6.8.0/examples/syncvalidation/

**Comment by jonlambert** [comment]
Disappointed to see no word still on this issue. I seem to be spending more and more time working around these quirks than just implementing without redux-form. 

In addition to triggering on submit, it would be really useful to be able to dispatch an action which triggers validation manually.

**Comment by twgraham** [comment]
This is an issue for me in wizard forms where I need to run field level validation before moving to the next step. When I try submit a given step to move to the next step on a pristine form, no validation is run. I can see how this might be an issue for any form though.

**Comment by danielrob** [comment]
A reminder that PR's are always strongly welcomed. 

Could someone explain what issue this actually causes? When fields are registered, validations are updated, setting errors on the form. When submitting, if there are validation errors then submission will fail. See for example https://codesandbox.io/s/8nk41wqrr8. So while UPDATE_SYNC_ERRORS is not called, the end goal seems to be achieved. 

Despite my brief input on this issue last year, I'm not sure I understand what the issue is here.

**Comment by AndyOGo** [comment]
@danielrob 
I can elaborate at least my specific case.
We implement a [Wizard](https://en.wikipedia.org/wiki/Wizard_(software)) leading the user through subsequent parts of a process.

Roughly the following happens:

1. fetch next process's step
2. [**Initially** create a form]
3. render process's interactive UI with the help of `redux-form`
4. set `initialValues` if any are provided
5. Once all inputs are valid allow **submission** through [`submit` action-creator](https://redux-form.com/7.3.0/docs/api/actioncreators.md/#-code-submit-form-string-code-)
   - To enable initialisation of values again for next step I triggered the [`reset` action-creator](https://redux-form.com/7.3.0/docs/api/actioncreators.md/#-code-reset-form-string-code-) upon success
   - Only the validation of the first steps work, all subsequent steps do not update the validation cache (only upon user interaction - i.e. changing a Field's value)
6. continue with 1 until the end is reached

**My Solution:**
I assumed my understanding of what the [`submit` action-creator](https://redux-form.com/7.3.0/docs/api/actioncreators.md/#-code-submit-form-string-code-) does is not what I expected it to do.
So first I tried to [`reset`](https://redux-form.com/7.3.0/docs/api/actioncreators.md/#-code-reset-form-string-code-) the form.
**But** as it says in the docs:
> Resets the values in the form back to the values past in with the most recent initialize action.
I figured, this does not seem to touch the validation checks at all. So went code-digging. And it turned out to be even worse, because:
- the result of [`isValid` selector](https://redux-form.com/7.3.0/docs/api/selectors.md/#-code-isvalid-formname-string-code-returns-code-state-gt-valid-boolean-code-) is cached
- within the [`validExceptSubmit ` prop](https://github.com/erikras/redux-form/blob/23b1fbd219bd1d81b55e2271c3352379c70007c4/src/createReduxForm.js#L797) which [is set here](https://github.com/erikras/redux-form/blob/23b1fbd219bd1d81b55e2271c3352379c70007c4/src/createReduxForm.js#L1005).
So I found this big hammer action-creator called [`destroy`](https://redux-form.com/7.3.0/docs/api/actioncreators.md/#-code-destroy-forms-string-code-), which makes subsequent validation checks work for my case:)
> Destroys the forms, removing all of their state.

Hope that clarifies and may help others.
IMHO I think in most case the cause of this "bug" is just plain miss-understanding of how `redux-form`'s API is designed to work. Though may it's possible that my initial expectation of how I think it work would suit better for this use case 🤔

**Comment by russellr86** [comment]
fwiw, I was experiencing the same issues after resetting a form... but calling initialize instead of reset did the trick for me.

**Comment by byjoh** [comment]
Thanks @russellr86.  I had the same issue where I was trying to reset my form and when the modal closed but then validation was broken the next time I re-opened it and tried to submit (reset seems to have removed the syncErrors array from the form node in redux??).  Using initialize instead works.

**Comment by exzizt** [comment]
It's nice to know it's an issue with the module... Thought I was going crazy!

**Comment by single-stop-justin** [comment]
yeah @exzizt , just running into this now. Wasted a few hours debugging.
